### PR TITLE
Authorise the API's using updated DB fields

### PR DESCRIPTION
**Pull request by gaurav-ahuja-synerzip**
- Authorise the API's using updated DB fields
- Code refactor

### CODE REVIEW

**Review by mangeshko — APPROVED** [review]

File: serverless.yml [hunk 1]
```diff
@@ -35,59 +35,68 @@ functions:
       - http:
           path: /groups
           method: post
+          cors: true
```
Comment (KarenKelly818) [inline serverless.yml:38]: Why are we enabling CORS?

**Comment by troy-rudolph** [comment]
@gaurav-ahuja-synerzip @KarenKelly818 @wilsonp68  

Hello.  There are changes here that enable CORS with no restrictions on the origin of the consumer.  This will make possible for browser-based applications to use SCIM-API.  The existing implementation (Python) does not do this.  Thus, it supports non-browser (back-end) intergration with the SCIM-API.  Before we enable brower-based application on the new SCIM implmentation (without any restrictions), I think we should at least discuss it.

Did some sort of CORS issue occur during testing?  In principle, the javacript that is generated by the doc page should have the same origin as the API itself and thus there should be no CORS issue since the browsers default to 'same orgin' when no CORS headers are provided.  

Please explain.  Thank you.

File: serverless.yml (hunk 1 above)
Comment (gaurav-ahuja-synerzip) [reply serverless.yml:38]: The swagger editor was throwing CORS error, that's why enabling the cors.

**Comment by gaurav-ahuja-synerzip** [comment]
@troy-rudolph @KarenKelly818 
we'll need to deploy the docs to the https://api-dev.myalchemy.com/scim2 if we don't want to enable the CORS.
I enabled the cors to make the swagger doc work on the swagger editor we are using.

**Comment by gaurav-ahuja-synerzip** [comment]
I have disabled the CORS.

**Review by KarenKelly818 — APPROVED** [review]

File: openAPI.yaml [hunk 2]
```diff
@@ -4,14 +4,7 @@ info:
...
 servers:
-  - url: /scim
-paths:
-  /:
-    get:
-      summary: get
-      responses:
-        '200':
-          description: '200'
+  - url: https://api-dev.myalchemy.com/scim2
```
Comment (troy-rudolph) [inline openAPI.yaml:7]: @gaurav-ahuja-synerzip @KarenKelly818

What is the plan to make the line above contain the value associated with the deployment location?  

Examples" 
local => ?
dev = "url: https://api-dev.myalchemy.com/scim2"
stage = "url: https://api-stage.myalchemy.com/scim2"
prod = "url: https://api.myalchemy.com/scim2"

**Review by troy-rudolph — CHANGES_REQUESTED** [review]

File: openAPI.yaml (hunk 2 above)
Comment (gaurav-ahuja-synerzip) [reply openAPI.yaml:7]: I'm working on deploying the swagger UI to the same URL for the API's then the URL can be updated from `https://api-dev.myalchemy.com/scim2` to just `/scim2`, then the swagger UI will use `${HOST}/scim2/` URL (HOST is where the swagger UI is deployed), resolves for all the environments.

**Review by troy-rudolph — APPROVED** [review]

**Pull request merged**